src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -44,9 +48,9 @@ public SearchResult<GuideSearchHit> search(@RestQuery @DefaultValue(QuarkusVersi
             @RestQuery String q,
             @RestQuery @DefaultValue("en") Language language,
             @RestQuery @DefaultValue("highlighted") String highlightCssClass,
-            @RestQuery @DefaultValue("0") int page,
-            @RestQuery @DefaultValue("1") int contentSnippets,
-            @RestQuery @DefaultValue("100") int contentSnippetsLength) {
+            @RestQuery @DefaultValue("0") @Min(0) int page,
+            @RestQuery @DefaultValue("1") @Range(min = 0, max = 10) int contentSnippets,
+            @RestQuery @DefaultValue("100") @Range(min = 0, max = 200) int contentSnippetsLength) {
```

Comment:
Maybe let's also have a message saying that it's not necessarily wrong to be over the max value but that it is just a limit for performance purposes.
Or if not a `message` in a constraint then a comment 😃

Reply:
Not sure the distinction matters to be honest.
I'm trying to avoid abuses here; the message is a nice bonus from HV but we could totally do without it.

Reply:
Ah, I just got what you meant by "comment". It would be useful, true, I'll add that.